.github/workflows/jupyterlite.yml

```diff
@@ -11,68 +11,69 @@ jobs:
-        run: python -m pip install jupyterlite[all] libarchive-c jupyter_server jupyterlab_server build pkginfo pyodide-build
-
-      - name: Setup EmSDK
-        # Doing this in a hacky script because the custom Action is requiring an explicit EmSDK version
-        # Pyodide is requiring a specific version too and it is changing regularly. This install the actual
-        # version required by the latest pyodide.
-
-        # Double initialization of PYODIDE_EMSCRIPTEN_VERSION: see https://github.com/pyodide/pyodide/issues/3430
         run: |
-          git clone https://github.com/emscripten-core/emsdk.git
-          cd emsdk
-          PYODIDE_EMSCRIPTEN_VERSION=$(pyodide config get emscripten_version)
-          PYODIDE_EMSCRIPTEN_VERSION=$(pyodide config get emscripten_version)
-          ./emsdk install ${PYODIDE_EMSCRIPTEN_VERSION}
-          ./emsdk activate ${PYODIDE_EMSCRIPTEN_VERSION}
-
-      - name: Build WASM wheels for frozenlist, coverage, and xDSL
+          python -m pip install jupyterlite[all] libarchive-c build pyodide-build
+          
+      - name: Build xDSL source distribution
         run: |
-          mkdir pypi
-          source emsdk/emsdk_env.sh
-          for i in frozenlist coverage
-          do
-
-          python -m pip download --no-binary :all: "$(grep $i requirements.txt)"
-          tar -xf $i-*.tar.gz
-          rm $i_*.tar.gz
-          cd $i-*
-          pyodide build
-          cd ..
-          cp $(find -path "./$i-*/dist/$i-*_wasm32.whl") pypi
+          cd xdsl
+          python setup.py sdist
 
-          done
+      - name: Restore cached Pyodide tree
+        id: cache-pyodide
+        uses: actions/cache@v3
+        with:
+          path: pyodide
+          key: pyodide
+          
+      - name: Clone pyodide if not cached
+        if: steps.cache-pyodide.outputs.cache-hit != 'true'
+        run: git clone https://github.com/pyodide/pyodide.git
 
-          pyodide build
-          cp $(find -path "./dist/xdsl-*.whl") pypi
+      - name: Build custom Pyodide distribution
+        run: |
+          XDSL_SOURCE_DIST=$(ls xdsl/dist)
+          XDSL_SOURCE_URL=file://$(pwd)/xdsl/dist/$XDSL_SOURCE_DIST
+          XDSL_SHA256=($(sha256sum xdsl/dist/$XDSL_SOURCE_DIST))
+          
+          cd pyodide
+          git fetch --all
+          git checkout 0.22.0a3
+          python -m pip install -r requirements.txt
+          sudo apt update && sudo apt install f2c
+          
+          rm packages/xdsl packages/frozenlist -rf
+          pyodide skeleton pypi xdsl
+          pyodide skeleton pypi frozenlist
+          
+          python -c 'import yaml; f = open("packages/xdsl/meta.yaml"); y = yaml.safe_load(f); f.close(); y["requirements"] = {"run" : ["frozenlist"]}; y["source"] = {"url" : "'$XDSL_SOURCE_URL'", "sha256" : "'$XDSL_SHA256'"}; f = open("packages/xdsl/meta.yaml", "w"); yaml.dump(y, f)'
+          
+          PYODIDE_PACKAGES="frozenlist,coverage,xdsl" make
```

Comment:
I don't understand everything written here and why it is necessary, but isn't there a github action for [pyodide](https://github.com/jobovy/pyodide-buildpackage-action)? Why not use that?

Reply:
The pyodide action does things similarly to how things worked before this PR.
Thing is, JupyterLite doesn't allow us to properly configure the code loading Pyodide on their side, so using that would require some hacking and rebuilding in JupyterLite.
Rebuilding Pyodide with xDSL in-tree is currently the cleanest solution (Even recommended [in their doc](https://jupyterlite.readthedocs.io/en/latest/howto/python/packages.html#bundling-additional-packages-by-default))